src/back/space/src/repo/table_space.cpp

```diff
@@ -0,0 +1,278 @@
+};
+
+std::vector<model::Space> Space::Select(const int offset, const int limit)
+{
+	auto res = _pg->Execute(storages::postgres::ClusterHostType::kMaster, kSelectSpace, offset, limit);
+
+	if (res.IsEmpty())
+	{
+		return {};
+	}
+
+	return res.AsContainer<std::vector<model::Space>>(pg::kRowTag);
+}
+
+const storages::postgres::Query kSelectSpaceAvailable{
+	R"~(
+		SELECT id, name, key, requestsAllowed, createdAt
+		FROM space
+		WHERE requestsAllowed = true
+		AND id NOT IN (
+			SELECT spaceId FROM space_user WHERE userId = $1
+		) OFFSET $2 LIMIT $3
+	)~",
+	storages::postgres::Query::Name{"select_space_available"},
+};
+
+std::vector<model::Space> Space::SelectAvailable(const std::string& userId, const int offset, const int limit)
+{
+	auto res = _pg->Execute(storages::postgres::ClusterHostType::kMaster, kSelectSpaceAvailable, userId, offset, limit);
+
+	if (res.IsEmpty())
+	{
+		return {};
+	}
+
+	return res.AsContainer<std::vector<model::Space>>(pg::kRowTag);
+}
+
+const storages::postgres::Query kSelectByUserId{
+	R"~(
+		SELECT id, name, key, requestsAllowed, createdAt
+		FROM space
+		WHERE id IN (
+			SELECT spaceId FROM space_user WHERE userId = $1
+		) OFFSET $2 LIMIT $3
+	)~",
+	storages::postgres::Query::Name{"select_space_by_user_id"},
+};
+
+std::vector<model::Space> Space::SelectByUserId(const std::string& userId, const int offset, const int limit)
+{
+	auto res = _pg->Execute(storages::postgres::ClusterHostType::kMaster, kSelectByUserId, userId, offset, limit);
+
+	if (res.IsEmpty())
+	{
+		return {};
+	}
+
+	return res.AsContainer<std::vector<model::Space>>(pg::kRowTag);
+}
+
+const storages::postgres::Query kCountSpace{
+	"SELECT count(id) FROM space",
+	storages::postgres::Query::Name{"count_space"},
+};
+
+int Space::Count() {
+	auto res = _pg->Execute(storages::postgres::ClusterHostType::kMaster, kCountSpace);
+
+	auto count = res.Front()[0].As<int64_t>();
+
+	return count;
+}
+
+const storages::postgres::Query kCountSpaceAvailable{
+	R"~(
+		SELECT count(id)
+		FROM space
+		WHERE requestsAllowed = true
+		AND id NOT IN (
+			SELECT spaceId FROM space_user WHERE userId = $1
+		)"
+	)~",
+	storages::postgres::Query::Name{"count_space_available"},
+};
+
+int Space::CountAvailable(const std::string& userId) {
+	auto res = _pg->Execute(storages::postgres::ClusterHostType::kMaster, kCountSpaceAvailable, userId);
+
+	auto count = res.Front()[0].As<int64_t>();
+
+	return count;
+}
+
+const storages::postgres::Query kCountByUserId{
+	R"~(
+		SELECT count(id)
+		FROM space
+		WHERE id IN (
+			SELECT spaceId FROM space_user WHERE userId = $1
+		)
+	)~",
+	storages::postgres::Query::Name{"count_space_by_user_id"},
+};
+
+int Space::CountByUserId(const std::string& userId) {
+	auto res = _pg->Execute(storages::postgres::ClusterHostType::kMaster, kCountByUserId, userId);
+
+	auto count = res.Front()[0].As<int64_t>();
+
+	return count;
+}
+
+const storages::postgres::Query kSelectSpaceByKey{
+	"SELECT * FROM space WHERE key=$1",
```

Comment:
Все поля получать не нужно если их не используем.
```suggestion
	"SELECT 1 FROM space WHERE key=$1",
```

Reply:
Готово.